PRICKLE4.py

```diff
@@ -0,0 +1,190 @@
+import os
+from abc import ABC, abstractmethod
+from Bio import SeqIO
+from Bio import SeqUtils
+
+
+class BiologicalSequence(ABC, str):
+
+    def __init__(self, seq):
+        self.seq = seq
+
+    #да, Никита уже сказал, что так лучше не делать, а оставить сдесь только абстрактные методы,
+    # но я зацепилась за идею не повторять проверки, раз суть для всех одна, только алфавиты разные,
+    # а переделать уже не успеваю
+    def check_alphabet(self):
+        if set(self.seq) <= self.alphabet:
+            raise UnexpectedSymbolInSeqError
+
+    # не поняла, не доделала
+    @abstractmethod
+    def to_print_seq(self):
+        return self.seq
+        pass

```

Comment:
Здесь, если я правильно понимаю, предполагалась реализация "вывода на печать в удобном виде"? В таком случае, стоило это сделать через __str__. Вообще, как я понимаю, предполагалось, что наш объект имеет атрибут, в котором лежит строка с последовательностью. Тогда можно было бы сделать так:
```suggestion
    def __str__(self):
        return self.seq
```
Как я понимаю, здесь логика была иной, учитывая наследование от строки. Такой объект тоже можно было бы перевести в тип строка, например, итерируясь по нему, добавляя элементы в list, а затем переведя лист в строку. Это не очень осмысленно, но если нужен именно тип строка, так можно было бы сделать.
Повторюсь только, что в абстрактном классе делать этого не стоило.